src/spi.rs

```diff
@@ -454,6 +454,26 @@ where
     }
 }
 
+impl<SPI, SCKPIN, MISOPIN, MOSIPIN> ::embedded_hal::spi::FullDuplex<u8>
+    for Spi<SPI, SCKPIN, MISOPIN, MOSIPIN, EightBit>
+where
+    SPI: Deref<Target = SpiRegisterBlock>,
+{
+    type Error = Error;
+
+    fn read(&mut self) -> nb::Result<u8, Error> {
+        self.check_read()?;
+        Ok(self.read_u8())
+    }
+
+    fn send(&mut self, byte: u8) -> nb::Result<(), Error> {
+        self.check_send()?;
+        self.send_u8(byte);
+        self.check_send().ok();
```

Comment:
Why are you discarding the result of the check_send the second time?

Reply:
I've simply reused the code from the `write` that does the same thing. You are right that it should raise any error instead of discarding it. Maybe this is also true in `write` (same file) ?

Reply:
I see, we should check the datasheet to see if the status register needs to be read after writing the data to the data register.
I believe that the propose of this construct is to check the status of errors such as Overrun, etc. so I propose that we should implement method `check_errors` that only checks for the errors and then propagate its result using the ? operator.

What do you think, @therealprof ?

Reply:
Looks like a good idea. If you think it's worth implementing, I can do it :) If you have other idea (eg. refactoring), I would be happy to help.

Reply:
Seems like a good plan, we might want to be careful about which errors we propagate and which ones we drop but we can look at that later when we have the `check_errors` function I guess.

Reply:
Ok, I'll draft something and add it here :)